Release process improvements and docs

hashhar opened the pull request.

hashhar committed:
Add .java-version file to specify JDK 11

This project requires JDK 11 to build and release.

hashhar committed:
Document release process concisely

losipiuk commented on README.md:
````diff
@@ -355,3 +355,42 @@ See [CHANGELOG](CHANGELOG.md) for details.
+  ./mvnw release:clean release:prepare -DpushChanges=false
+  ./mvnw release:perform -DlocalCheckout=true
+  ```
+- Once the above finishes you'll have 2 commits locally. Create a new pull
````
Typicallly I just merge those - without a PR

hashhar replied:
How? I can't seem to push to the `develop` branch directly.

losipiuk replied:
Maybe it is because I have org admin

hashhar replied:
I'll keep the pull request process here then for the time being until I can think of some solution.

hashhar committed:
Change server id in distributionManagement to ossrh

ossrh is what all Sonatype documentation uses and also credentials for
both snapshot and staging repositories are common.

hashhar committed:
Use a single repository entry for OSSRH and change server id to ossrh

OSSRH exposes repository groups which doesn't require multiple
repository entries. This allows us to converge on just ossrh as the
server id making it easy to manage by just having a single entry in
settings.xml.

This is also the canonical way how most open-source Maven projects
configure their pom.

hashhar commented:
AC. Fixed the build failure using the first two commits.

PTAL @losipiuk
